src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * The Parser class is responsible for parsing user input and determining the corresponding actions to take.
+ * It provides methods to read user input, check for specific command types, and manage the input scanner.
+ */
+public class Parser {
+    static String userInput;
+    static Scanner scanner = new Scanner(System.in);
+
+    /**
+     * Retrieves the user input from the scanner.
+     *
+     * @return The user input as a string.
+     */
+    public String getUserInput() {
+        return scanner.nextLine();
+    }
+
+    /**
+     * Sets the user input to a new value.
+     *
+     * @param newUserInput The new user input value to set.
+     */
+    public void setUserInput(String newUserInput) {
+        userInput = newUserInput;
+    }
+
+    /**
+     * Checks if the user wants to exit the application.
+     *
+     * @return True if the user input is "bye", indicating an exit command; false otherwise.
+     */
+    public boolean bye() {
```

Comment:
I feel like the method names here are a little misleading as well; even though the names are verbs but they sound more like they will perform the action of listing, marking etc. instead of returning a boolean. Maybe something like `isBye()` will be more suitable?